pkg/controller/controller_utils.go

```diff
@@ -942,16 +942,6 @@ func PatchNodeTaints(c clientset.Interface, nodeName string, oldNode *v1.Node, n
 		return fmt.Errorf("failed to marshal old node %#v for node %q: %v", oldNode, nodeName, err)
 	}
 
-	newTaints := newNode.Spec.Taints
```

Comment:
Yes, but this would add implicit assumption that this function needs to get a copy of an object, which can lead to bugs if used somewhere else. I'm against such change without a proof that it noticeably improves performance.